pkg/kopialib/storage/s3.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2024 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+
+	"github.com/kanisterio/kanister/pkg/kopialib"
+	"github.com/kanisterio/kanister/pkg/utils"
+	"github.com/kopia/kopia/repo/blob"
+	"github.com/kopia/kopia/repo/blob/s3"
+)
+
+var _ Storage = &s3Storage{}
+
+var requiredS3Arguments = []string{
+	kopialib.BucketKey,
+}
+
+type s3Storage struct {
+	Options *s3.Options
+	Create  bool
```

Comment:
what does `Create` decide/configure here, create if not exists?
Also, connect would connect to an S3 repo?

Reply:
Create the S3 bucket if it does not exist

Reply:
https://github.com/kanisterio/kanister/pull/2606#discussion_r1483910217